pkg/virt-api/webhooks/validating-webhook/admitters/vmi-create-admitter.go

```diff
@@ -950,9 +960,25 @@ func ValidateVirtualMachineInstanceMandatoryFields(field *k8sfield.Path, spec *v
 	return causes
 }
 
-func ValidateVirtualMachineInstanceMetadata(field *k8sfield.Path, metadata *metav1.ObjectMeta, config *virtconfig.ClusterConfig) []metav1.StatusCause {
+func ValidateVirtualMachineInstanceMetadata(field *k8sfield.Path, metadata *metav1.ObjectMeta, config *virtconfig.ClusterConfig, accountName string) []metav1.StatusCause {
+
 	var causes []metav1.StatusCause
 	annotations := metadata.Annotations
+	labels := metadata.Labels
+	filteredLabels, hasRestrictedLabels := FilterKubevirtLabels(labels)
+
+	// Validate kubevirt.io labels presence. Restricted labels allowed
+	// to be created only by known service accounts
+	if len(filteredLabels) > 0 && hasRestrictedLabels {
+		allowed := GetAllowedServiceAccounts()
+		if _, ok := allowed[accountName]; !ok {
+			causes = append(causes, metav1.StatusCause{
+				Type:    metav1.CauseTypeFieldValueNotSupported,
+				Message: "creation of kubevirt.io/ labels on a VMI object is restricted",
```

Comment:
We don't restrict all labels... just a subset of `kubevirt.io/` for now.
Also, I think that it is worth telling the user exactly which label he added which is restricted. Otherwise, it can be a frustrating experience.

Reply:
+1 for including the label we restrict.

Reply:
@danielBelenky @xpivarc Lets just document the restriction over the usage of system reserved labels. Filtering the exact ones that were changed by the user will take more effort.